homeassistant/components/utility_meter/sensor.py

```diff
@@ -240,9 +249,17 @@ def __init__(
         tariff_entity,
         tariff,
         unique_id,
+        legacy_meter_name=None,
     ):
         """Initialize the Utility Meter sensor."""
         self._attr_unique_id = unique_id
+        if legacy_meter_name:
+            slug = (
+                slugify(f"{parent_meter} {tariff}")
```

Comment:
Why do we check for the `legacy_meter_name`, but base the entity_id on the meter + tariff?

Reply:
because that is the current inconsistent behaviour I'm trying to avoid breaking through this PR